### DISCUSSION TITLE
Prevent search bar from stealing focus from active window

### DISCUSSION THREAD

**Discussion by archetyped** [espanso/espanso#1254] [Feature requests and Ideas]
**Is your feature request related to a problem? Please describe.**
The active window loses focus when the Espanso's search bar is displayed.  This causes issues when trying to use Espanso's output in dialogs, etc. that are closed/hidden when the window loses focus (example: Firefox's Add Bookmark dialog), or in apps where text is reformatted when focus is lost.

**Describe the solution you'd like**
The active window should remain focused when Espanso's search bar is displayed.

Example: [PhraseExpress'](https://www.phraseexpress.com/) phrase menu does not take focus from the active window, allowing text insertion in dialogs and without triggering auto-reformatting in apps.

**Additional context**
Example of dialog that closed when the active window's focus is lost: Firefox Add Bookmark dialog
![Firefox_Dialog_Add-Bookmarkpng](https://user-images.githubusercontent.com/3237917/150695540-eb8ee371-215d-4b18-858d-3a45d9b920b5.png)

**Comment by federico-terzi** [comment]
Hey @archetyped,

Thank you for the feedback! Good point, I'll investigate :) This one will be very hard to pull off in a cross-platform way

Cheers :)

**Reply by xazw** [reply]
Somewhat off-topic Federico, but I've noticed and really love your great and welcoming attitude towards everyone in the open-source community. Thanks so much for your great software, and for being a kind and positive influence on the world.

**Reply by iamqiz** [reply]
This is a feature I want all the time

**Reply by sanathusk** [reply]
Hi @federico-terzi  i would  request to to bring back the capability to focus auto focus the input in searchbox based on a flag in user config, i have many triggers defined and i use the searchbox to find the right one, a i do most of my trying in Vcode, so each time the seearch box is launched, the input does not have the focus and i need to use the mouse to get the focus, if you miss to use mouse and select, the content that i type is added inside the vscode file.
Let me know if you are open to PR, will try to implement the same after discussion.
